src/Symfony/Component/Console/Formatter/Lexer.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Formatter;
+
+use Symfony\Component\Console\Exception\FormatterTooLargeInputException;
+use Symfony\Component\Console\Formatter\Tokens\EosToken;
+use Symfony\Component\Console\Formatter\Tokens\FullTagToken;
+use Symfony\Component\Console\Formatter\Tokens\FullTextToken;
+use Symfony\Component\Console\Formatter\Tokens\SeparatorToken;
+use Symfony\Component\Console\Formatter\Tokens\WordToken;
+use Symfony\Component\Console\Helper\Helper;
+
+/**
+ * Tokenize a custom text to handle styling and wrapping it.
+ *
+ * @author Krisztián Ferenczi <ferenczi.krisztian@gmail.com>
+ */
+class Lexer implements LexerInterface
+{
+    /** @var FullTextToken */
+    protected $fullTextToken;
```

Comment:
everything must use private visibility, not protected one, unless we *explicitly* decide to make inheritance the supported extension point (hint: such decision is always the last choice).